### PR TITLE
New: Onboarding [ED-6175, PRDH-871]

### PR DESCRIPTION
**Pull request by udidol**

### CODE REVIEW

File: core/app/modules/onboarding/assets/js/pages/hello-theme.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+	 * and skipped.
+	 */
+	useEffect( () => {
+		if ( ! helloInstalledInOnboarding && state.isHelloThemeActivated ) {
+			const stateToUpdate = getStateObjectToUpdate( state, 'steps', pageId, 'completed' );
+
+			updateState( stateToUpdate );
+
+			goToNextScreen();
+		}
```
Comment (yossi-elementor) [inline core/app/modules/onboarding/assets/js/pages/hello-theme.js:41]: return () => {
resetScreenContent
}

File: core/app/modules/onboarding/assets/js/pages/site-logo.js [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+		setUploadImageAjax( {
+			data: {
+				action: 'elementor_upload_site_logo',
+				fileToUpload: fileToUpload,
+			},
+		} );
+	}, [] );
+
+	const dismissUnfilteredFilesCallback = useCallback( () => {
```
Comment (yossi-elementor) [inline core/app/modules/onboarding/assets/js/pages/site-logo.js:93]: Avoid usecallback

**Review by yossi-elementor — CHANGES_REQUESTED** [review]

File: core/app/assets/js/molecules/upload-file.js [hunk 3]
```diff
@@ -22,26 +25,54 @@ export default function UploadFile( props ) {
...
+
+							// Initialize the WP Media frame.
+							frame = wp.media( {
+								multiple: false,
+								library: {
+									type: [ 'image', 'image/svg+xml' ],
+								},
+							} );
+
+							frame.on( 'select', () => props.onWpMediaSelect( frame ) );
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/assets/js/molecules/upload-file.js:72]: Done

File: core/app/modules/onboarding/assets/js/components/button.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+export default function Button( props ) {
+	const { button, type } = props;
+
+	let buttonClasses = 'e-onboarding__button';
+
+	if ( type ) {
+		buttonClasses += ` e-onboarding__button-${ type }`;
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/components/button.js:7]: Turns out there is a place where I use the button without passing a `type`

File: core/app/modules/onboarding/assets/js/components/button.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+export default function Button( props ) {
+	const { button, type } = props;
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/components/button.js:2]: Done

File: core/app/modules/onboarding/assets/js/components/card.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+export default function Card( props ) {
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/components/card.js:1]: Done

File: core/app/modules/onboarding/assets/js/components/checkbox-with-label.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+			/>
+			{ props.labelText }
+		</label>
+	);
+}
+
+CheckBoxWithLabel.propTypes = {
+	checked: PropTypes.any,
+	labelText: PropTypes.string,
+	onChangeCallback: PropTypes.func,
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/components/checkbox-with-label.js:19]: Done

File: core/app/modules/onboarding/assets/js/components/go-pro-popover.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
+import { Context } from '../context/context';
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/components/go-pro-popover.js:1]: Done

File: core/app/modules/onboarding/assets/js/components/go-pro-popover.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+			elementorCommon.events.dispatchEvent( {
+				placement: elementorAppConfig.onboarding.eventPlacement,
+				event: 'already have pro',
+				step: state.currentStep,
+			} );
+
+			// Open the Pro Upload screen in a popup.
+			window.open(
+				jQuery( getProButton ).attr( 'href' ) + '&mode=popup',
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/components/go-pro-popover.js:29]: Done

File: core/app/modules/onboarding/assets/js/components/go-pro-popover.js [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+			// Run the callback for when the upload succeeds.
+			elementorCommon.elements.$window
+				.on( 'elementor/upload-and-install-pro/success/', () => {
+					updateState( { hasPro: true } );
+				} );
+		} );
+	}, [] );
+
+	const goProButton = props.buttons.find( ( button ) => 'go-pro' === button.id ),
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/components/go-pro-popover.js:42]: Done something similar, added a comment above this line explaining it

File: core/app/modules/onboarding/assets/js/components/go-pro-popover.js [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+			onClick: () => {
+				elementorCommon.events.dispatchEvent( {
+					placement: elementorAppConfig.onboarding.eventPlacement,
+					event: 'get elementor pro',
+					step: state.currentStep,
+				} );
+			},
+		};
+
+	elementorCommon.elements.$window
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/components/go-pro-popover.js:57]: Deleted this one, it's not needed.

File: core/app/modules/onboarding/assets/js/components/layout/footer-buttons.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+
+	let classNames = 'e-onboarding__footer';
+
+	if ( props.className ) {
+		classNames += ' ' + props.className;
+	}
+
+	return (
+		<Grid container alignItems="center" justify="space-between" className={ classNames }>
+			{ maybeGetActionButton() }
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/components/layout/footer-buttons.js:30]: Done

File: core/app/modules/onboarding/assets/js/components/layout/header.js [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+	usePageTitle( { title: props.title } );
+
+	const { state } = useContext( Context );
+
+	const onClose = () => {
+		elementorCommon.events.dispatchEvent( {
+			placement: elementorAppConfig.onboarding.eventPlacement,
+			event: 'close modal',
+			step: state.currentStep,
+			connect_id: '',
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/components/layout/header.js:18]: Done

File: core/app/modules/onboarding/assets/js/components/layout/header.scss [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+
+	&-logo {
+
+		.eps-app__logo {
+			background-color: $platform-darkest-gray;
+			color: $white;
+			width: 1.3rem;
+			height: 1.3rem;
+			line-height: 1.3rem;
+			font-size: calc(0.4 * 1.2rem);
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/components/layout/header.scss:11]: Removed calculation

File: core/app/modules/onboarding/assets/js/components/layout/layout.js [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+					placement: elementorAppConfig.onboarding.eventPlacement,
+					event: 'go pro',
+					step: state.currentStep,
+					contributor: state.isUsageDataShared,
+				} );
+			},
+		} );
+	}
+
+	useEffect( () => {
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/components/layout/layout.js:66]: Done

File: core/app/modules/onboarding/assets/js/components/progress-bar/progress-bar.js [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+import ProgressBarItem from './progress-bar-item';
+
+export default function ProgressBar() {
+	const { state } = useContext( Context ),
+		navigate = useNavigate(),
+		progressBarItemsConfig = [
+			{
+				id: 'account',
+				title: __( 'Elementor Account', 'elementor' ),
+				route: '',
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/components/progress-bar/progress-bar.js:14]: Done

File: core/app/modules/onboarding/assets/js/context/context.js [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+				siteName: false,
+				siteLogo: false,
+				goodToGo: false,
+			},
+		},
+		[ state, setState ] = useState( initialState ),
+		updateState = useCallback( ( newState ) => {
+			setState( ( prev ) => ( { ...prev, ...newState } ) );
+		}, [ setState ] ),
+		getStateObjectToUpdate = useCallback( ( stateObject, mainChangedPropertyKey, subChangedPropertyKey, subChangedPropertyValue ) => {
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/context/context.js:32]: Done

File: core/app/modules/onboarding/assets/js/pages/account.scss [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+.e-onboarding__already-have-account-text {
+	margin-top: 24px;
+	width: 325px;
+	text-align: center;
+
+	a {
+		text-decoration: underline;
+		color: #353298;
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/pages/account.scss:15]: Done

File: core/app/modules/onboarding/assets/js/pages/hello-theme.js (hunk 1 above)
Comment (udidol) [reply core/app/modules/onboarding/assets/js/pages/hello-theme.js:41]: I checked and this function does not match here

File: core/app/modules/onboarding/assets/js/pages/site-logo.js [hunk 19]
```diff
@@ -0,0 +1,293 @@
...
+			data: {
+				action: 'elementor_update_site_logo',
+				data: JSON.stringify( {
+					attachmentId: file.id,
+				} ),
+			},
+		} );
+	}, [ file ] );
+
+	const uploadSiteLogo = useCallback( ( fileToUpload ) => {
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/pages/site-logo.js:82]: Done

File: core/app/modules/onboarding/assets/js/pages/site-logo.js [hunk 20]
```diff
@@ -0,0 +1,293 @@
...
+							heading={ __( 'Drop image here', 'elementor' ) }
+							secondaryText={ __( 'or', 'elementor' ) }
+							buttonText={ __( 'Open Media Library', 'elementor' ) }
+							buttonVariant="outlined"
+							buttonColor="cta"
+							icon={''}
+							type="wp-media"
+							filetypes={ [ 'jpg', 'jpeg', 'png', 'svg' ] }
+							onFileSelect={ ( selectedFile ) => {
+								setFileSource( 'drop' );
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/pages/site-logo.js:221]: Done

File: core/app/modules/onboarding/assets/js/pages/site-name.js [hunk 21]
```diff
@@ -0,0 +1,112 @@
...
+			text: __( 'Next', 'elementor' ),
+			onClick: () => {
+				elementorCommon.events.dispatchEvent( {
+					placement: elementorAppConfig.onboarding.eventPlacement,
+					event: 'next',
+					step: state.currentStep,
+				} );
+
+				// Only run the site name update AJAX if the new name is different than the existing one and it isn't empty.
+				if ( actionButtonRef.current.value !== state.siteName && '' !== actionButtonRef.current.value ) {
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/pages/site-name.js:28]: Done

File: core/app/modules/onboarding/assets/js/utils/connect.js [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
+
+		elementorCommon.config.library_connect.is_connected = true;
+		elementorCommon.config.library_connect.current_access_level = data.kits_access_level || data.access_level || 0;
+
+		stateToUpdate.isLibraryConnected = true;
+
+		updateState( stateToUpdate );
+	};
+
+	if ( props.buttonRef ) {
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/assets/js/utils/connect.js:18]: Done

File: core/common/assets/js/common.js [hunk 23]
```diff
@@ -6,6 +6,8 @@ import Finder from 'elementor-common-modules/finder/assets/js/finder';
 import Connect from 'elementor-common-modules/connect/assets/js/connect';
 import API from './api/';
 import WordpressComponent from 'elementor-api/components/wordpress/component';
+import EventsDispatcherComponent from 'elementor-common-modules/event-tracker/assets/js/data/component';
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/common/assets/js/common.js:9]: Commented out

File: core/common/modules/event-tracker/data/controller.php [hunk 24]
```diff
@@ -0,0 +1,58 @@
+<?php
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/common/modules/event-tracker/data/controller.php:1]: It's off 100%

**Review by yossi-elementor — APPROVED** [review]

**Pull request merged**